### PR TITLE
Add scalar AArch64 Keccak-f1600 ASM

### LINKED ISSUE #132: Develop X18-free scalar AArch64-Keccak leveraging lazy rotation
https://eprint.iacr.org/2022/1243 describes a spill-free implementation of Keccak-f1600 for AArch64. It is available on [pqax](https://github.com/slothy-optimizer/pqax).

However, this implementation uses `x18` and is thus not portable -- specifically, it does not work on Mac.

This task is to consider whether simple adjustments can be made to the implementation of https://eprint.iacr.org/2022/1243.pdf so it avoids x18 at the cost of a very small number of spills. If feasible, this would be a good starting point for integrating into MLKEM-C-AArch64. Otherwise, we may obtain _a_ version with spills from SLOTHY, but experiments so far suggest that it may not be optimal.

### PR DESCRIPTION
**Pull request by hanno-becker**
Fixes #132

### CODE REVIEW

File: fips202/asm/aarch64/keccak_f1600_x1_scalar_pre_opt.S [hunk 1]
```diff
@@ -0,0 +1,934 @@
...
+        // bic X<tmp>, X<Bbo>, X<Bbi>, ror #42                  // .............................*.........................
+        // eor x6, X<tmp>, X<Bbe>, ror #41                      // ..............................*........................
+        // bic X<tmp>, X<Bbu>, X<Bbo>, ror #57                  // .....................................*.................
+        // eor x11, X<tmp>, X<Bbi>, ror #35                     // .........................................*.............
+        // bic X<tmp>, X<Bba>, X<Bbu>, ror #50                  // ...................................*...................
+        // eor x16, X<tmp>, X<Bbo>, ror #43                     // ........................................*..............
+        // bic X<tmp>, X<Bbe>, X<Bba>, ror #44                  // ...............................*.......................
+        // eor x21, X<tmp>, X<Bbu>, ror #30                     // ....................................*..................
+        // eor x1, x1, X<cur_const>                             // .................................................*.....
```
Comment (cothan) [inline fips202/asm/aarch64/keccak_f1600_x1_scalar_pre_opt.S:918]: I see these lines comment out, I think it's only relevant to SLOTHY dev, can you delete them ?

File: fips202/asm/aarch64/keccak_f1600_x1_scalar_pre_opt.S [hunk 2]
```diff
@@ -0,0 +1,934 @@
...
+        // eor x1, X<Bba>, X<tmp>, ror #21               // ..............................................*......
+        // bic X<tmp>, X<Bbo>, X<Bbi>, ror #42           // ................................................*....
+        // eor x6, X<tmp>, X<Bbe>, ror #41               // ..................................................*..
+        // bic X<tmp>, X<Bbu>, X<Bbo>, ror #57           // ...............................................*.....
+        // eor x11, X<tmp>, X<Bbi>, ror #35              // ................................................*....
+        // bic X<tmp>, X<Bba>, X<Bbu>, ror #50           // ..................................................*..
+        // eor x16, X<tmp>, X<Bbo>, ror #43              // ....................................................*
+        // bic X<tmp>, X<Bbe>, X<Bba>, ror #44           // .................................................*...
+        // eor x21, X<tmp>, X<Bbu>, ror #30              // ...................................................*.
+        // eor x1, x1, X<cur_const>                      // ....................................................*
```
Comment (cothan) [inline fips202/asm/aarch64/keccak_f1600_x1_scalar_pre_opt.S:674]: I see these lines comment out, I think it's only relevant to SLOTHY dev, can you delete them ?

**Review by cothan — CHANGES_REQUESTED** [review]

File: fips202/asm/aarch64/keccak_f1600_x1_scalar_pre_opt.S (hunk 1 above)
Comment (hanno-becker) [reply fips202/asm/aarch64/keccak_f1600_x1_scalar_pre_opt.S:918]: I'd like to keep the 'clean' SLOTHY origin somewhere, but you're right that the auto-generated SLOTHY comments may not be suitable. I'll remove them for now and consider where else to add it.

File: fips202/asm/aarch64/keccak_f1600_x1_scalar_pre_opt.S (hunk 2 above)
Comment (hanno-becker) [reply fips202/asm/aarch64/keccak_f1600_x1_scalar_pre_opt.S:674]: Same

**Comment by hanno-becker** [comment]
Fixes #132

**Review by cothan — APPROVED** [review]

**Pull request merged**